src/test/java/com/napier/sem/DAO_UnitTests.java

```diff
@@ -459,13 +459,130 @@ public void topNCapitalCitiesIn_negativeNListEmpty()
     public void topNCapitalCitiesIn_happyPath()
     {
         // given
-        String areaFilter = App.COUNTRY;
-        String areaName = "France";
+        String areaFilter = App.REGION;
+        String areaName = "Nordic Countries";
 
         // then
         dao.topNCapitalCitiesIn(areaFilter, areaName, 15); // No Error
     }
 
+    /**
+     * Unit tests covering the DAO.populationLivingInAndNotInCities method
+     */
+    // test that populationResidenceReport list is empty when areaFilter and areaName are null
+    @Test
+    public void populationLivingInAndNotInCities_bothArgumentsNullListEmpty()
+    {
+        // given
+        String areaFilter = null;
+        String areaName = null;
+
+        // when
+        ArrayList<Record> populationResidenceReport = dao.populationLivingInAndNotInCities(areaFilter, areaName);
+
+        // then
+        assertEquals(0, populationResidenceReport.size());
+    }
+
+    // test that populationResidenceReport list is empty when areaFilter is null
+    @Test
+    public void populationLivingInAndNotInCities_areaFilterNullListEmpty()
+    {
+        // given
+        String areaFilter = null;
+
+        // when
+        ArrayList<Record> populationResidenceReport = dao.populationLivingInAndNotInCities(areaFilter, "Scotland");
+
+        // then
+        assertEquals(0, populationResidenceReport.size());
+    }
+
+    // test that populationResidenceReport list is empty when areaName is null
+    @Test
+    public void populationLivingInAndNotInCities_areaNameNullListEmpty()
+    {
+        // given
+        String areaName = null;
+
+        // when
+        ArrayList<Record> populationResidenceReport = dao.populationLivingInAndNotInCities(App.COUNTRY, areaName);
+
+        // then
+        assertEquals(0, populationResidenceReport.size());
+    }
+
+    // check test passes with valid areaFilter and areaName
+    @Test
+    public void populationLivingInAndNotInCities_happyPath()
+    {
+        // given
+        String areaFilter = App.CONTINENT;
+        String areaName = "Asia";
+
+        // then
+        dao.populationLivingInAndNotInCities(areaFilter, areaName); // No Error
+    }
+
+    //TODO: Test only continent, region or country can be entered.
```

Comment:
@scan-lan I think you said you were going to look at this, I can't remember if I added the comment or if someone else did but are you happy for it to be removed?

Reply:
we can remove it now, i've made a ticket for it